bundles/org.openhab.binding.tacmi/src/main/java/org/openhab/binding/tacmi/internal/schema/ApiPageParser.java

```diff
@@ -368,45 +384,68 @@ private void getApiPageEntry(@Nullable String id2, int line, int col, String sho
                         String val = valParts[0].replace(',', '.');
                         float bd = Float.parseFloat(val);
                         if (valParts.length == 2) {
-                            if ("°C".equals(valParts[1])) {
-                                channelType = "Number:Temperature";
-                                state = new QuantityType<>(bd, SIUnits.CELSIUS);
-                            } else if ("%".equals(valParts[1])) {
-                                // channelType = "Number:Percent"; Number:Percent is currently not handled...
-                                channelType = "Number:Dimensionless";
-                                state = new QuantityType<>(bd, Units.PERCENT);
-                            } else if ("Imp".equals(valParts[1])) {
-                                // impulses - no idea how to map this to something useful here?
+                            var unitStr = valParts[1];
+                            var unitData = taCmiSchemaHandler.unitsCache.get(unitStr);
+                            if (unitData == null) {
+                                // we try to lookup the unit given by TA.
+                                try {
+                                    Unit<?> parsedUnit = SimpleUnitFormat.getInstance().parse(unitStr);
```

Comment:
Hi @lsiepel - this is where I access `SimpleUnitFormat` to find the appropriate unit for the string we received from the TA equipment. If there is a better way to look up units, please give me a pointer ;)

Reply:
Hi. I Tried it with `Units.getInstance().getUnit()` but it didn't resolve the units properly...